src/components/types.ts

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import type meta from '../../icons/meta'
+
+export type StringWithAutocomplete<T> = T | (string & Record<never, never>)
+
+export type IconNames = keyof (typeof meta)['icons']
```

Comment:
How about we put this in `icons/meta.ts`? That's where the value is and its utility is not only for components. And then how about we clear up the name of this file, since it's not all types in components as they are usually defined with each component. How about `utlity.ts` or `utility_types.ts` or something in that nature?